### PR TITLE
Implement the payment logic

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

### CODE REVIEW

File: backends/hospital-service/main.bal [hunk 1]
```diff
@@ -228,10 +293,7 @@ service on ln {
             returns Status|http:NotFound|error {
         lock {
             if appointments.hasKey(appointment_id) {
-                string dob = appointments.get(appointment_id).patient.dob;
-                string yob = regexp:split(re `-`, dob)[0];
-                string currYear = regexp:split(re `-`, time:utcToString(time:utcNow()))[0];
-                int age = check int:fromString(currYear) - check int:fromString(yob);
+                int age = check getAge( appointments.get(appointment_id).patient.dob);
```
Comment (dulajdilshan) [inline backends/hospital-service/main.bal]:
```suggestion
                int age = check getAge(appointments.get(appointment_id).patient.dob);
```

File: backends/hospital-service/main.bal [hunk 2]
```diff
@@ -118,10 +141,52 @@ service /healthcare on ln {
...
     }
 
-    // todo: payments 
+    resource function get payments() returns Payment[] {
+        lock {
+            return <readonly> from Payment payment in payments select payment;
+        }
+    }
+
+    resource function get payments/payment/[string payment_id]() returns Payment|http:NotFound {
```
Comment (dulajdilshan) [inline backends/hospital-service/main.bal:182]: `/payments/payment/<payment_id>` feels a little bit off the REST best practices. How about using `/payments/<payment_id>`?

If it is to identify this as a separate service,Let's go with the current.

Comment (MaryamZi) [reply backends/hospital-service/main.bal:182]: Right now, this is a 1:1 match with WSO2 backend. IMO, we can do any and all significant changes later, once we have all the tutorials.

Comment (dulajdilshan) [reply backends/hospital-service/main.bal:182]: Ack

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Pull request merged**